pre_processing/setup.F90

```diff
@@ -2055,7 +2191,7 @@ subroutine setup_imager(args, opts, source_attributes, channel_info, verbose)
    implicit none
 
    type(setup_args_t),        intent(inout) :: args
-   type(preproc_opts_t),      intent(in)    :: opts
+   type(preproc_opts_t),      intent(inout) :: opts
```

Comment:
I don't think is right. `opts%channel_ids` is an array read from the driver file where the user states what channels they want to preprocess, which can be some subset of those available in the file. You're overriding that functionality for Python reads and Python reads only.

I think you can allow the normal behaviour by using `common_setup` and using the read from the Python file in the same way that the hard-coded arrays are used for the other instruments.

Reply:
From memory this was the only way I could think of to get it working. Changes welcome.